.github/workflows/test-deploy-main.yml

```diff
@@ -296,3 +296,30 @@ jobs:
           USERNAME: ${{ secrets.USERNAME }}
           NEW_PASSWORD: ${{ secrets.NEW_PASSWORD }}
           VALID_PASSWORD: ${{ secrets.VALID_PASSWORD }}
+
+  publish:
+    runs-on: ubuntu-latest
```

Comment:
Should we only publish to next when all tests pass?

Reply:
Oh man, good catch! Let me make this more explicit.

Reply:
Done here [6782fe1](https://github.com/aws-amplify/amplify-ui/pull/1746/commits/6782fe17d2c768194dae31f68b82757963b4932e)

Reply:
Awesome!